### ISSUE TITLE
VNC Does not work on runpod

### ISSUE THREAD

**Issue by chicagotaco** [asyafiqe/Rope-docker-headless-vnc-container#1]
I am unable to connect to the VNC server when using Runpod. I can only use noVNC

**Comment by Kornifikator** [comment]
I can confirm that it doesn't work at all although I followed the instructions for runpod. Shame. 
Maybe the dev will update and fix it. Would be amazing

**Comment by asyafiqe** [comment]
It seems that Runpod doesn't support traffic on ports other than HTTP. noVNC is working on Runpod. VNC is functioning on my local system.

**Comment by Kornifikator** [comment]
Thank you for your reply. However, unfortunately I have to say that noVNC doesn't work either. I tried everything and followed all the steps. I had no success to get things running.


> It seems that Runpod doesn't support traffic on ports other than HTTP. noVNC is working on Runpod. VNC is functioning on my local system.

**Comment by asyafiqe** [comment]
I just deployed in Runpod and it was okay. Can you try this image: asyafiqe/rope_vnc:231002

**Comment by Kornifikator** [comment]
Thanks for your efforts. I will try it tonight. Right now, I can't.
By the way, I saw that all repo files are from last year, but since then there have been updates to Rope. Are you planning to update to the new version?


> I just deployed in Runpod and it was okay. Can you try this image: asyafiqe/rope_vnc:231002

**Comment by asyafiqe** [comment]
I just update the repo and docker images to Rope Pearl-00. Jupyter may not be able to create a new file in runpod. For upload/download file, you can use filebrowser on port 8585. All working well  in vast.ai.

For VNC in runpod, put the port 5901 in TCP.

**Comment by Kornifikator** [comment]
Hi, it is me again. Sorry for the big delay. I had some rl issues that took a lot of time and attention...
Anyway, I followed all your steps, tried your latest repo and docker images (including this image too: asyafiqe/rope_vnc:231002) and nothing worked out on runpod.
In the following I will share my observations:
- Firstly, after the start of the Pod, sometimes there aren't any Port buttons. But when I restart the Pod they suddenly appear. By clicking them, a new empty tab opens up. That's it. Moreover, the Port buttons disappear again after this one-time-click.
- Secondly, I added the Pod log file to this answer. Maybe it gives you a hint of what is going wrong:
[logs (1).txt](https://github.com/user-attachments/files/17475420/logs.1.txt)

I really don't understand why it doesn't work. Any solution would be great


**EDIT:** I was able to run your template on Vast without any problems. Thanks for the great work!



> I just update the repo and docker images to Rope Pearl-00. Jupyter may not be able to create a new file in runpod. For upload/download file, you can use filebrowser on port 8585. All working well in vast.ai.
> 
> For VNC in runpod, put the port 5901 in TCP.
